app/store/fetching_thread_store.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {BehaviorSubject} from 'rxjs';
+
+type State = {[id: string]: boolean};
+
+const defaultState: State = {};
+
+export const subject: BehaviorSubject<State> = new BehaviorSubject(defaultState);
+
+export const setFetchingThreadState = (rootId: string, isFetching: boolean) => {
+    const prevState = subject.value;
+    subject.next({
+        ...prevState,
```

Comment:
This will grow the object in an unbounded way. Probably not a big deal, but if we travel through many threads without closing the app this may become big. 0/5 whether you should delete the element from the state when `isFetching` is false.